service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java

```diff
@@ -52,18 +53,18 @@ public static PayonePostServiceImpl of(final String payoneServerApiUrl) throws I
     public Map<String, String> executePost(final BaseRequest baseRequest) throws PayoneException {
 
         try {
-            Map<String, Object> mappedListParameters = getObjectMapWithExpandedLists(baseRequest.toStringMap(false));
+            List<BasicNameValuePair> mappedListParameters =
+                    getNameValuePairsWithExpandedLists(baseRequest.toStringMap(false));
+
+            String serverResponse = executePostRequestToString(this.serverAPIURL, mappedListParameters);
 
-            String serverResponse = Unirest.post(this.serverAPIURL)
-                    .fields(mappedListParameters)
-                    .asString().getBody();
             if (serverResponse.contains("status=ERROR")) {
                 LOG.error("-> Payone POST request parameters: "
-                        + getObjectMapWithExpandedLists(baseRequest.toStringMap(true)).toString());
+                        + getNameValuePairsWithExpandedLists(baseRequest.toStringMap(true)).toString());
```

Comment:
I think if in general throughout the code you use string interpolation (with `String.format`) then use it consistently and avoid using concatenation.

Reply:
0c52f8de